core/frontend/src/PlanarClipMaskState.ts

```diff
@@ -69,29 +64,51 @@ export class PlanarClipMaskState {
     return this._allLoaded ? this._tileTreeRefs : undefined;
   }
 
-  public getPlanarClipMaskSymbologyOverrides(): FeatureSymbology.Overrides | undefined {
+  public getPlanarClipMaskSymbologyOverrides(view: ViewState3d, context: SceneContext): FeatureSymbology.Overrides | undefined {
+    let overrideModels;
+    if (view.isSpatialView())
+      overrideModels = view.getMaskModels(this.settings.modelIds, PlanarClipMaskMode.Priority === this.settings.mode);
+
+    if (!this.settings.subCategoryOrElementIds && !overrideModels)
+      return undefined;
+
+    const overrides = new FeatureSymbology.Overrides();
+
+    if (overrideModels) {
+      // overrideModels is used for batched models.  For those, we need to create model overrides for visibility (using transparency).
+      const appOn = FeatureAppearance.fromTransparency(0.0);
+      const appOff = FeatureAppearance.fromTransparency(1.0);
+      // For Priority or Models mode, we need to start with the current overrides and modify them
+      if (PlanarClipMaskMode.Priority === this.settings.mode || PlanarClipMaskMode.Models === this.settings.mode) {
+        const curOverrides = new FeatureSymbology.Overrides(context.viewport);
```

Comment:
Why did you decide to get rid of the `RenderTarget.getCurrentSymbologyOverrides` method you previously had, and instead recreate the overrides from scratch here?

Reply:
Because we needed to modify them, and don't want to modify the target one.